core/types/block.go

```diff
@@ -232,6 +232,9 @@ type Block struct {
 	// inter-peer block relay.
 	ReceivedAt   time.Time
 	ReceivedFrom interface{}
+
+	// blobs provides DA check
+	blobs BlobTxSidecars
```

Comment:
Weren't sidecars supposed to be together with block only for propagation purposes? Why it got added inside the block itself?

Reply:
The block struct may be a good glue struct, we can set blobs into it, and transfer them to the related component. There are other extra fields in here to record metadata.

We also need another extra blob with the block for p2p messages.

Reply:
Just wondering if there's any process that is done with blocks passing apart from broadcasting. If there is any such process then it will slow down the work due to block becoming very heavy due to blobs. 

> We also need another extra blob with the block for p2p messages.

If we are including sidecar inside block definition itself then we won't need to make any changes for the p2p messages and the current mechanism only will need to check for Cancun or not and act accordingly.

Reply:
Ok, we can discuss more about the definition. The block could transfer to `InsertChain`, `IsDataAvailbility` for checking block and blob, block is a glue struct, it contains header, block body, blobs, and other metadata. So we can remove receovedBlobCache too.

Block is passed by pointer, so it will cost little in most scenarios. It is only be used in `NewBlockMsg`, but only header, body will be encoded.

```go
func (b *Block) EncodeRLP(w io.Writer) error {
	return rlp.Encode(w, &extblock{
		Header:      b.header,
		Txs:         b.transactions,
		Uncles:      b.uncles,
		Withdrawals: b.withdrawals,
	})
}
```

Reply:
> It is only be used in NewBlockMsg, but only header, body will be encoded.

If we encode the sidecars as well then we may not even need to modify `BlockWithExtraData` and have this definition of block everywhere except when it comes to storing. As storing of sidecars and blocks will be separate. What do you think?